File: src/main/java/com/speechkids/controller/AuthController.java
package com.speechkids.controller;

import com.speechkids.dto.AuthResponse;
import com.speechkids.dto.LoginRequest;
import com.speechkids.dto.RefreshRequest;
import com.speechkids.dto.RegisterRequest;
import com.speechkids.dto.UserDto;
import com.speechkids.entity.User;
import com.speechkids.exception.NotFoundException;
import com.speechkids.repository.UserRepository;
import com.speechkids.service.AuthService;
import com.speechkids.service.UserPrincipal;
import jakarta.validation.Valid;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/auth")
public class AuthController {
    private final AuthService authService;
    private final UserRepository userRepository;

    public AuthController(AuthService authService, UserRepository userRepository) {
        this.authService = authService;
        this.userRepository = userRepository;
    }

    @PostMapping("/register")
    public AuthResponse register(@Valid @RequestBody RegisterRequest request) {
        return authService.register(request);
    }

    @PostMapping("/login")
    public AuthResponse login(@Valid @RequestBody LoginRequest request) {
        return authService.login(request);
    }

    @PostMapping("/refresh")
    public AuthResponse refresh(@Valid @RequestBody RefreshRequest request) {
        return authService.refresh(request);
    }

    @PostMapping("/logout")
    public ResponseEntity<Void> logout(@Valid @RequestBody RefreshRequest request) {
        authService.logout(request.refreshToken());
        return ResponseEntity.noContent().build();
    }

    @GetMapping("/me")
    public UserDto me(@AuthenticationPrincipal UserPrincipal principal) {
        User user = userRepository.findById(principal.getId())
                .orElseThrow(() -> new NotFoundException("User not found"));
        return authService.toUserDto(user);
    }
}

File: auth-service/bin/main/com/halfi/controller/AuthController.kt
package com.halfi.auth.controller

import com.halfi.auth.dto.AuthResponse
import com.halfi.auth.dto.RegisterRequest
import com.halfi.auth.service.AuthService
import com.halfi.dto.response.LoginResponse
import org.springframework.http.HttpHeaders
import org.springframework.http.ResponseCookie
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import jakarta.servlet.http.HttpServletResponse

data class EmailRequest(val email: String)
data class ResetPasswordRequest(val email: String, val code: String, val newPassword: String)

@RestController
@RequestMapping("/api/auth")
class AuthController(private val authService: AuthService) {

    @PostMapping("/register")
    fun register(@RequestBody request: RegisterRequest): AuthResponse {
        return authService.register(request)
    }

    @PostMapping("/resend-code")
    fun resendCode(@RequestBody request: EmailRequest): AuthResponse {
        return authService.resendCode(request.email)
    }

    @PostMapping("/forgot-password")
    fun forgotPassword(@RequestBody request: EmailRequest): AuthResponse {
        return authService.forgotPassword(request.email)
    }

    @PostMapping("/reset-password")
    fun resetPassword(@RequestBody request: ResetPasswordRequest): AuthResponse {
        return authService.resetPassword(request.email, request.code, request.newPassword)
    }

    @PostMapping("/verify")
    fun verify(@RequestParam email: String, @RequestParam code: String): ResponseEntity<String> {
        val isVerified = authService.verifyCode(email, code)
        return if (isVerified) {
            ResponseEntity.ok("Email verified successfully. You can now login.")
        } else {
            ResponseEntity.badRequest().body("Invalid or expired code")
        }
    }

    @GetMapping("/test")
    fun test() = "Auth Service is up and running!"

    @PostMapping("/login")
    fun login(
        @RequestBody request: RegisterRequest, 
        response: HttpServletResponse
    ): LoginResponse {
        val loginResponse = authService.login(request)
        
        val cookie = ResponseCookie.from("refreshToken", loginResponse.refreshToken)
            .httpOnly(true)
            .secure(false) // В production нужно исправить на true для HTTPS
            .path("/")
            .maxAge(30L * 24 * 60 * 60) // 30 дней
            .build()
            
        response.addHeader(HttpHeaders.SET_COOKIE, cookie.toString())
        
        return loginResponse
    }
}